src/shared/bus-unit-util.c

```diff
@@ -1306,6 +1306,17 @@ int bus_append_unit_property_assignment(sd_bus_message *m, const char *assignmen
 
                 r = sd_bus_message_append(m, "v", "t", t);
 
+        } else if (streq(field, "OnCalendar")) {
+                CalendarSpec *spec = NULL;
+
+                r = calendar_spec_from_string(eq, &spec);
+                if (r < 0)
+                        return log_error_errno(r, "Invalid calendar spec %s: %m", eq);
+
+                calendar_spec_free(spec);
```

Comment:
hmm, so validity checks are mostly redundant, the server has to validate this anyway, and the server errors are pretty much the same level of useful as client side errors.

I mean, parsing (and thus validating) user input on the client side makes sense if we have to do that in order to be able to marshal things properly in the bus message, but if we don't need to do that the server-side message is as good as the client-side message.

In particular for things like enum strings we pass to the server this has the benefit that older clients can make use of new enum values, without needing update... and in a way this applies here too: should we extend the calendar spec language one day it might be nice if clients could just make use of this even if they are older than the server side...

Reply:
Yeah, this is redundant. Dropped.